contrib/nydus-backend-proxy/src/main.rs

```diff
@@ -0,0 +1,240 @@
+                                read += n as u64;
+                                if unread < BUFSIZE {
+                                let part = &allbuf[0..unread];
+                                yield io::Cursor::new(part.to_vec());
+                                }
+                                else{
+                                    yield io::Cursor::new(allbuf.to_vec());
+                                }
+                            }
+                            Err(_err) =>break
+                        }
+                    }
+                })
+                .raw_header("Docker-Content-Digest", self.dis)
+                .raw_header("Content-Type", "application/octet-stream")
+                .raw_header("Content-Range", rangestring)
+                .ok()
+        }
+    }
+
+    #[derive(Responder)]
+    pub enum StoredData {
+        AllFile(Option<FileStream>),
+        Range(RangeStream),
+    }
+
+    #[get("/<_namespace>/<_repo>/blobs/<digest>")]
+    pub async fn blob_fetch(
+        _namespace: PathBuf,
+        _repo: PathBuf,
+        digest: String,
+        headerdata: HeaderData,
+    ) -> Result<StoredData, Status> {
+        let dis = &digest[7..];
+        //if no range in Request header,return fetch blob response
+        if headerdata.range.is_empty() {
+            let mut filepath = IMAGEPATH.clone();
+            filepath.push(&dis);
+            return Ok(StoredData::AllFile(
+                NamedFile::open(filepath)
+                    .await
+                    .ok()
+                    .map(|nf| FileStream(nf, dis.to_string())),
+            ));
+        } else {
+            let blobfd = match BLOBFDMAP.get(&ffi::OsString::from(&dis)) {
+                Some(fd) => fd,
+                None => return Err(Status::NotFound),
+            };
+            let metadata = match blobfd.metadata() {
+                Ok(meta) => meta,
+                Err(_err) => {
+                    return Err(Status::InternalServerError);
+                }
+            };
+            let ranges = match HttpRange::parse(&headerdata.range, metadata.len()) {
+                Ok(r) => r,
+                Err(_err) => {
+                    return Err(Status::RangeNotSatisfiable);
+                }
+            };
+            let startpos = ranges[0].start as u64;
+            let size = ranges[0].length;
+            return Ok(StoredData::Range(RangeStream {
+                dis: dis.to_string(),
+                len: size,
+                start: startpos,
+            }));
+        }
+    }
+}
+
+#[rocket::main]
+async fn main() {
+    let args: Vec<String> = env::args().collect();
+    if let Err(_e) = rocket::build()
+        .mount(
+            "/",
+            rocket::routes![getblob::blob_check, getblob::blob_fetch],
+        )
+        .mount("/", FileServer::from(&args[1]))
+        .launch()
+        .await
+    {
+        println!("Whoops! Rocket didn't launch!");
```

Comment:
The `err` should be printed to stderr using something like `panic!()`, and exit with non-zero status.